static/js/app.js

```diff
@@ -663,3 +665,109 @@ faqQuestion.forEach(item => {
+    
+  },
+  next: async ()=> {   
+
+    // Grab data necessary to update settings:onboarding_state value
+    const onboardingContainer = document.querySelector(".c-multipart-premium-onboarding");
+    const mainContainer = document.getElementById("profile-main");
+    const profileId = mainContainer.dataset.profileId;
+    let currentOnboardingState = parseInt(onboardingContainer.dataset.onboardingCompletedStep, 10);
+    
+    // Bump onboarding to next interger
+    currentOnboardingState++
+
+    const settings = {
+      onboarding_state: currentOnboardingState,
+    };
+
+    // Show next step content
+    const activeOnboardingSlide = document.querySelector(".c-premium-onboarding-step.is-visible");
+    activeOnboardingSlide.classList.remove("is-visible");
+    activeOnboardingSlide.nextElementSibling.classList.add("is-visible");
+    
+    // Show next step buttons
+    const activeOnboardingActions = document.querySelector(".c-premium-onboarding-actions.is-visible");
+    activeOnboardingActions.classList.remove("is-visible");
+    activeOnboardingActions.nextElementSibling.classList.add("is-visible");
+        
+    // Update progress bar
+    const activeOnboardingProgressSlides = document.querySelectorAll(".c-premium-onboarding-progress-bar-item.is-completed");
+    // Get most recent "completed" progress bar section
+    const activeOnboardingProgressSlide = activeOnboardingProgressSlides[activeOnboardingProgressSlides.length - 1];
+    activeOnboardingProgressSlide.nextElementSibling.classList.add(
+      "is-completed"
+    );
+
+    try {
+        const response = await patchProfile(profileId, settings);
+        if (!response.ok) {
+            throw new Error("Immediately catch'd to show an error message.");
+        }
+        
+        // Update DOM data-set for future usage
+        onboardingContainer.dataset.onboardingCompletedStep = currentOnboardingState;
+        
+    } catch (e) {
+        // saveError.classList.remove("hidden");
+    }
+
+  },
+  quit: async (e)=> {
+     
+    if (e.target.classList.contains("js-premium-onboarding-skip-step")) {
+      const onboardingContainer = document.querySelector(".c-multipart-premium-onboarding");
+      const onboardingCurrentStep = parseInt(onboardingContainer.dataset.onboardingCompletedStep, 10) + 1;
+      sendGaPing("Premium Onboarding", "Engage", "onboarding-skip", onboardingCurrentStep);
+    }
+
+    const onboardingContainer = document.querySelector(".c-multipart-premium-onboarding");
+    let maxOnboardingState = parseInt(onboardingContainer.dataset.maxOnboardingAvailable, 10);
+    
+    const mainContainer = document.getElementById("profile-main");
+    mainContainer.classList.remove("is-premium-onboarding");
+    
+    const profileId = mainContainer.dataset.profileId;
+    const settings = {
+      onboarding_state: maxOnboardingState,
+    };
+
+    try {
+        const response = await patchProfile(profileId, settings);
```

Comment:
`profileData`

Reply:
Revised in a1c3f191dab5eea3763ab1663160911d233cb1c5